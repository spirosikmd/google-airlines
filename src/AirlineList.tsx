import React from 'react';
import Airline from './Airline';
import { AirlineData } from './airlines';

const AirlineList = ({ airlines }: { airlines: AirlineData[] }) => (
  <div className="pt-6">
    <div className="flex flex-wrap">
      {airlines.map(airline => (
        <Airline key={airline.id} airline={airline} />
      ))}
    </div>
  </div>
);

export default AirlineList;
