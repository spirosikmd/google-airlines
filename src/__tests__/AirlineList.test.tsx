import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AirlineList from '../AirlineList';

afterEach(cleanup);

test('renders a list of airlines', () => {
  const airlines = [
    { id: 'iata1_airline1', iata: 'iata1', airline: 'airline1' },
    { id: 'iata2_airline2', iata: 'iata2', airline: 'airline2' }
  ];

  const { asFragment } = render(<AirlineList airlines={airlines} />);

  expect(asFragment()).toMatchSnapshot();
});
