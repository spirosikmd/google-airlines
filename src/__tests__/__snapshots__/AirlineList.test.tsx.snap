// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of airlines 1`] = `
<DocumentFragment>
  <div
    class="pt-6"
  >
    <div
      class="flex flex-wrap"
    >
      <div
        class="p-6 m-2 h-20 w-20 border border-solid rounded-full border-gray-200 hover:border-purple-600 flex items-center justify-center"
        data-tip="airline1"
        tabindex="0"
      >
        <img
          alt="airline1"
          class="max-w-full"
          src="data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' width%3D'200' height%3D'150' viewBox%3D'0 0 200 150'%2F%3E"
          style="background: rgb(229, 252, 245);"
        />
      </div>
      <div
        class="p-6 m-2 h-20 w-20 border border-solid rounded-full border-gray-200 hover:border-purple-600 flex items-center justify-center"
        data-tip="airline2"
        tabindex="0"
      >
        <img
          alt="airline2"
          class="max-w-full"
          src="data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' width%3D'200' height%3D'150' viewBox%3D'0 0 200 150'%2F%3E"
          style="background: rgb(229, 252, 245);"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
